frodo.js

```diff
@@ -1,54 +1,49 @@
 /**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.